src/dotnet-compile-fsc/project.json

```diff
@@ -3,9 +3,14 @@
   "compilationOptions": {
     "emitEntryPoint": true
   },
+  "compile": [
+    "**/*.cs",
+    "../dotnet/CommandLine/*.cs",
+    "../dotnet/commands/dotnet-compile-csc/AssemblyInfoOptionsCommandLine.cs",
```

Comment:
No, these classes can't be shared in Compiler.Common.  That is what got us into trouble in the first place. These members were exposing System.CommandLine types, which isn't a real supported product, just like the internal classes in `src/dotnet/CommandLine` aren't a real supported product. When we get real `CommandLine` APIs we can revisit this decision, but for now forking is going to be the best way forward.

From my notes in the PR description:
F# compiler shares the CommandLine files and two common compiler utility files with the C# compiler. This is needed to keep our CommandLine APIs internal. If/when F# moves to a different repo, these files will need to be copied.